revup/amend.py

```diff
@@ -46,6 +46,27 @@ def invoke_editor_for_commit_msg(
     return re.sub(r"^\s*#.*$", "", msg, flags=re.M).strip()
 
 
+async def get_topic_summary(args: argparse.Namespace, git_ctx: git.Git) -> str:
+    topics = topic_stack.TopicStack(
+        git_ctx,
+        args.base_branch,
+        args.relative_branch,
+        None,
+        None,
+    )
+
+    try:
+        await topics.populate_topics()
```

Comment:
populate_topics takes a decent bit of time, can we

- cache the result of populate_topics from parse_ref_or_topic so it only gets called once
- only get summary if args.parse_topics is true

Reply:
Both of these requests are added. While I was testing pulling up the editor with the commit was not noticeably slow for this repo, or our internal repo.

Reply:
yeah i only mean slow in a relative sense. it takes about .2s which is significant in the sense that we want the tool to be as fast as reasonably possible